src/plugins/ieo/modules/public/sagas/ieoItemSaga.ts

```diff
@@ -0,0 +1,35 @@
+// tslint:disable-next-line
+import { call, put } from 'redux-saga/effects';
+import { API, RequestOptions } from '../../../../../api';
+import { pluginsList } from '../../../../../api/config';
+import {
+    FetchIEOItem,
+    ieoItemData,
+    ieoItemError,
+} from '../actions';
+
+const requestOptions: RequestOptions = {
+    apiVersion: 'applogic',
+    withHeaders: false,
+};
+
+export function* ieoItemSaga(action: FetchIEOItem) {
+    try {
+        const data = yield call(API.get(requestOptions), `/public/ieo/sales/${action.payload}`);
+        const ieoPlugin = pluginsList().find(item => item.name === 'ieo');
+
+        if (ieoPlugin && ieoPlugin.config.metadata) {
+            const details = yield call(
+                API.get(requestOptions),
+                `/public/metadata/search?key=IEO-${data.currency_id}-${data.id}`,
+            );
+            const detailsData = JSON.parse(details.value);
+
+            yield put(ieoItemData({ ieo: data, details: detailsData }));
```

Comment:
maybe it`s better to store details as string, and parse in containers (if here can be any data).